### DISCUSSION TITLE
Help setting up with clangd and cmake

### DISCUSSION THREAD

**Discussion by djmittens** [jrouwe/JoltPhysics#1220] [Q&A] [answered]
Hello, I am looking to try using jolt in my game. However ive run into a problem with setting up LSP for cmake and jolt.  When i go to definition on any of the classes from jolt, it seems like clangd doesnt respect the imports around `Jolt.h` causing too many errors to be emitted for the library files.  Since i use it as a cmake subdirectory, how can i fix this ?
![image](https://github.com/user-attachments/assets/fb24d8ca-d962-447d-8507-cf8ef71ae0f2)

**Comment by jrouwe** [comment]
If I understand you correctly, the issue is only that in your editor when viewing Jolt files you see too many errors.

If you can configure your editor to always load Jolt.h before loading the file then these errors should go away.

**Reply by djmittens** [reply]
Thank you for quick response ! 
Yep i understand i guess i dont know how to do that, im using clangd as the LSP, which relies on `compile_commands.json`  generated by CMake.  I dont know how that can be configured to do this include specifically for Jolt.  I guess im asking for how to configure cmake specifically for clangd  https://clangd.llvm.org/installation.html 😅

**Comment by djmittens** [comment, marked as answer]
In case someone else runs into this problem as well, I was able to configure clangd to attach this header to any jolt file found in the project, through an include flag.  This is the `.clangd` config that fixed this for me:
```yaml
CompileFlags:
  CompilationDatabase: "build"
---
If:
  PathMatch: [.*Jolt/.*\.h, .*Jolt/.*\.cpp]
CompileFlags:
  Add: "-includeJolt/Jolt.h"
```
but i think its a bit hacky, im hoping for a more general solution with cmake, or maybe some instructions for setup in the main documentation.

Edit: 
This doesn't seem to fix all includes, there are some cases left where the imports are insufficient eg. not all imports used are specified in some of  the files. But at least its good enough for most code navigation.
